subsys/bluetooth/mesh/sensor.c

```diff
@@ -91,9 +91,17 @@ bool bt_mesh_sensor_delta_threshold(const struct bt_mesh_sensor *sensor,
 	 */
 	if (sensor->state.threshold.delta.type ==
 	    BT_MESH_SENSOR_DELTA_PERCENT) {
-		int64_t prev_mill = llabs(SENSOR_MILL(&sensor->state.prev));
-
-		thrsh_mill = (prev_mill * thrsh_mill) / (100LL * 1000000LL);
+		/* Store the Status Delta Trigger threshold in sensor_value first to avoid int64_t
+		 * overflow caused by multiplication of values converted by the SENSOR_MILL() macro.
+		 */
+		struct sensor_value thrsh_delta = delta_mill >= 0 ?
```

Comment:
Use a pointer here, perhaps? I realize the struct is quite tiny so it wouldn't make a big difference

Reply:
Yeah, pointer will cause dereferencing, so not sure it is better.